data_source.go

```diff
@@ -421,6 +422,11 @@ func (ds *AnySource) ConfigurePulseLengths(nsamp, npre int) error {
 	return nil
 }
 
+// SetCoupling is a no-op for generic data sources
+func (ds *AnySource) SetCoupling(status CouplingStatus) error {
```

Comment:
this should return an error like "setting coupling for AnySourc not yet implemented"

Reply:
Trigger coupling is not ever going to be implemented on generic sources, because they don't have error-like "auxiliary channels" to go with each primary channel.

However, you might be right that this ought to return an error (other than not-implemented-yet). Let me think about it. Yeah, I guess that's right. Let's make this return error, thereby eliminating the special case in the RPC server.